cmd/id-exporter/main.go

```diff
@@ -193,18 +193,43 @@ Required arguments:
 - config
 - outfile`
 
-func main() {
-	outFile := flag.String("outfile", "", "File to write contacts to (defaults to stdout).")
-	grace := flag.Duration("grace", 2*24*time.Hour, "Include contacts with certificates that expired in < grace ago")
-	domainsFile := flag.String("domains", "", "If provided only output contacts for certificates that contain at least one of the domains in the provided file. Provided file should contain one domain per line")
-	withExampleHostnames := flag.Bool("with-example-hostnames", false, "In addition to IDs, gather an example domain name that corresponds to that ID")
-	type config struct {
-		ContactExporter struct {
-			DB cmd.DBConfig
-			cmd.PasswordConfig
-			Features map[string]bool
+// unmarshalHostnames unmarshals a hostnames file and ensures that the file
+// contained at least one entry.
+func unmarshalHostnames(filePath string) ([]string, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	scanner.Split(bufio.ScanLines)
+
+	var hostnames []string
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.Contains(line, " ") {
+			return nil, fmt.Errorf(
+				"line: %q contains more than one entry, entries must be separated by newlines", line)
 		}
+		hostnames = append(hostnames, line)
 	}
+
+	if len(hostnames) == 0 {
+		return nil, errors.New("provided file contains 0 hostnames")
+	}
+	return hostnames, nil
+}
+
+func main() {
+	outFile := flag.String("outfile", "", "File to output results JSON to.")
+	grace := flag.Duration("grace", 2*24*time.Hour, "Include results with certificates that expired in < grace ago.")
+	hostnamesFile := flag.String(
+		"hostnames", "", "Only include results with unexpired certificates that contain hostnames\nlisted (newline separated) in this file.")
+	withExampleHostnames := flag.Bool(
+		"with-example-hostnames", false, "Include an example hostname for each registration ID with an unexpired certificate.")
+	useDefaultIsolationLevel := flag.Bool(
+		"use-default-isolation-level", false, "Use database default transaction isolation level. (default is READ UNCOMMITTED)")
```

Comment:
Nit for others to comment on: I think it's unlikely MariaDB is going to change the default tx-isolation, but do we want to maybe rephrase this to not assume something about that?

Reply:
This is actually meant to be quite the opposite. If this flag isn't set, `id-exporter` defaults to setting the Tx isolation level as `READ UNCOMMITTED`

```
Usage of id-exporter:
  -config string
    	File containing a JSON config.
  -grace duration
    	Include results with certificates that expired in < grace ago. (default 48h0m0s)
  -hostnames string
    	Only include results with unexpired certificates that contain hostnames
    	listed (newline separated) in this file.
  -outfile string
    	File to output results JSON to.
  -use-default-isolation-level
    	Use database default transaction isolation level. (default is READ UNCOMMITTED)
  -with-example-hostnames
    	Include an example hostname for each registration ID with an unexpired certificate.
```

Reply:
My point is that `default is READ UNCOMMITTED` is dependent on MariaDB making no changes.

Reply:
oh DOH

Reply:
![bushes](https://user-images.githubusercontent.com/518542/119205791-2374ba80-ba4e-11eb-8d8d-be728a73a858.gif)

Reply:
😆 this language might just be confusing. Is there a better way for me to communicate this in the flag descriptor?

Reply:
I think it's the double-use of "default" there that is most confusing to me. Maybe:

```suggestion
		"use-default-isolation-level", false, "Do not override database transaction isolation level to READ UNCOMMITTED")
```

Reply:
Thank you!

Reply:
Shipped a commit with this change, please re-approve.